api/v1alpha1/shared_types.go

```diff
@@ -609,3 +609,35 @@ type HTTP2Settings struct {
 	// +optional
 	OnInvalidMessage *InvalidMessageAction `json:"onInvalidMessage,omitempty"`
 }
+
+// CustomErrorResponse defines the configuration to override specific responses with a custom one.
+type CustomErrorResponse struct {
+	// Match configuration.
+	Match CustomErrorResponseMatch `json:"match"`
+	// Response configuration.
+	Response CustomResponse `json:"response"`
+}
+
+// CustomErrorResponseMatch defines the configuration for matching a user response to return a custom one.
+type CustomErrorResponseMatch struct {
+	// Status codes to match on. A match is successful if any of the status codes match.
+	StatusCodes []string `json:"statusCodes"`
```

Comment:
* should we support values like `4xx` ?
* should we support ranges like `500-520` ?

Reply:
Yeah, some way to provide a range is definitely important since 4xx/5xx will be a very common use-case

Reply:
I think status classes like `4xx` and `5xx` are good enough for now. I can't think of any use cases like `500-520`, but if they do exist, we can add that later.

Reply:
ive gone ahead and added `range`